lmod/openmpi_custom.lua

```diff
@@ -99,8 +99,13 @@ elseif  ompiv == "3.1" or ompiv == "4.0" or ompiv == "4.1" then
 	        if ompiv == "3.1" then -- removed in 4.0; don't use ofi by default for cuda
 			setenv("OMPI_MCA_mtl", "^mxm,ofi")
 			setenv("OMPI_MCA_oob", "^ud")
-		elseif ompiv == "4.1" then -- omnipath nodes may run out of contexts if the ofi btl is enabled
-			setenv("OMPI_MCA_btl", "^openib,ofi")
+		else
+			if interconnect == "ethernet" then
+				setenv("OMPI_MCA_mtl", "^ofi")
+			end
+			if ompiv == "4.1" then -- omnipath nodes may run out of contexts if the ofi btl is enabled
```

Comment:
The change of imbrication level for the ompiv == "4.1" will break when we get to support ompiv 5.x, won't it ?

Reply:
i have no idea about 5.0 but for now this block is inside a big
`elseif  ompiv == "3.1" or ompiv == "4.0" or ompiv == "4.1" then`
block, so once 5.0 is out and we have it this will need adjusting anyway.